src/main/scala/com.snowplowanalytics.iglu/ctl/commands/Generate.scala

```diff
@@ -246,27 +254,41 @@ object Generate {
                                      splitProduct: Boolean,
                                      noHeader: Boolean,
                                      owner: Option[String])
-                                    (schemas: NonEmptyList[SelfDescribingSchema[JValue]]): DdlOutput = {
+                                    (schemas: NonEmptyList[IgluSchema]): DdlOutput = {
     // Build table definitions from JSON Schemas
-    val validatedDdls = schemas.toList.map(schema => selfDescSchemaToDdl(schema, dbSchema, splitProduct, owner, varcharSize, noHeader).map(ddl => (schema.self, ddl)))
-    val (errors, ddlPairs) = validatedDdls.separate
+    val ddlPairs = schemas.toList.map(schema => (schema.self, selfDescSchemaToDdl(schema, dbSchema, splitProduct, owner, varcharSize, noHeader)))
     val ddlMap = groupWithLast(ddlPairs)
 
     // Build migrations and order-related data
-    val migrationMap = buildMigrationMap(schemas.toList)
-    val validOrderingMap = Migration.getOrdering(migrationMap)
-    val orderingMap = validOrderingMap.collect { case (k, Validated.Valid(v)) => (k, v) }
-    val (_, migrations) = Migrations.reifyMigrationMap(migrationMap, Some(dbSchema), varcharSize).separate
+    val orderingMap: Map[ModelGroup, List[String]] = createOrderingMap(schemas)
+
+    val migrationMap = Try(buildMigrationMap(schemas.toList)).getOrElse(Map.empty)
```

Comment:
Isn't `getOrElse(Map.empty)` just hide an invalid state?

Reply:
Yeah, I just put it as temporary. I think I should check invalid cases instead of just using try.